components/client/typescript/src/schemas/stacks/payload.ts

```diff
@@ -0,0 +1,125 @@
+import { Static, Type } from '@sinclair/typebox';
+import {
+  BlockIdentifierSchema,
+  Nullable,
+  OperationIdentifierSchema,
+  TransactionIdentifierSchema,
+} from '../common';
+import { StacksTransactionEventSchema } from './tx_events';
+import { StacksTransactionKindSchema } from './tx_kind';
+
+export const StacksPrincipalSchema = Type.String();
+export type StacksPrincipal = Static<typeof StacksPrincipalSchema>;
+
+export const StacksExecutionCostSchema = Type.Optional(
+  Type.Object({
+    read_count: Type.Integer(),
+    read_length: Type.Integer(),
+    runtime: Type.Integer(),
+    write_count: Type.Integer(),
+    write_length: Type.Integer(),
+  })
+);
+export type StacksExecutionCost = Static<typeof StacksExecutionCostSchema>;
+
+export const StacksTransactionReceiptSchema = Type.Object({
+  contract_calls_stack: Type.Array(Type.String()),
+  events: Type.Array(StacksTransactionEventSchema),
+  mutated_assets_radius: Type.Array(Type.String()),
+  mutated_contracts_radius: Type.Array(Type.String()),
+});
+export type StacksTransactionReceipt = Static<typeof StacksTransactionReceiptSchema>;
+
+export const StacksTransactionPositionSchema = Type.Object({
+  index: Type.Integer(),
+  micro_block_identifier: Type.Optional(BlockIdentifierSchema),
+});
+export type StacksTransactionPosition = Static<typeof StacksTransactionPositionSchema>;
+
+export const StacksTransactionMetadataSchema = Type.Object({
+  description: Type.String(),
+  execution_cost: StacksExecutionCostSchema,
+  fee: Type.Integer(),
+  kind: StacksTransactionKindSchema,
+  nonce: Type.Integer(),
+  position: StacksTransactionPositionSchema,
+  proof: Nullable(Type.String()),
+  raw_tx: Type.String(),
+  receipt: StacksTransactionReceiptSchema,
+  result: Type.String(),
+  sender: StacksPrincipalSchema,
+  sponsor: Type.Optional(StacksPrincipalSchema),
+  success: Type.Boolean(),
+});
+export type StacksTransactionMetadata = Static<typeof StacksTransactionMetadataSchema>;
+
+export const StacksOperationAccountSchema = Type.Object({
+  address: StacksPrincipalSchema,
+  sub_account: Type.Optional(StacksPrincipalSchema),
+});
+export type StacksOperationAccount = Static<typeof StacksOperationAccountSchema>;
+
+export const StacksOperationAmountSchema = Type.Object({
+  currency: Type.Object({
+    decimals: Type.Integer(),
+    symbol: Type.String(),
+    metadata: Type.Object({
+      asset_class_identifier: Type.String(),
+      asset_identifier: Nullable(Type.String()),
```

Comment:
When do we Nullable vs Optional?

Reply:
I based these on the actual Rush implementation... Nullable is when the key can actually be sent as `null`, optional is when it can be ommitted